va/x11/va_dri3.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "va_dri3.h"
+
+Pixmap VA_DRI3_create_pixmap(Display *dpy,
+                             Drawable draw,
+                             int width, int height, int depth,
+                             int fd, int bpp, int stride, int size)
+{
+    xcb_connection_t *c = XGetXCBConnection(dpy);
+    if (fd >= 0) {
+        xcb_pixmap_t pixmap = xcb_generate_id(c);
+        xcb_dri3_pixmap_from_buffer(c, pixmap, draw, size, width, height,
+                                    stride, depth, bpp, fd);
+        return pixmap;
+    }
+    return 0;
+}
+
+int VA_DRI3_create_fd(Display *dpy, Pixmap pixmap, int *stride)
+{
+    xcb_connection_t *c = XGetXCBConnection(dpy);
+    xcb_dri3_buffer_from_pixmap_cookie_t cookie;
+    xcb_dri3_buffer_from_pixmap_reply_t *reply;
+
+    cookie = xcb_dri3_buffer_from_pixmap(c, pixmap);
+    reply = xcb_dri3_buffer_from_pixmap_reply(c, cookie, NULL);
+    if (!reply) {
+        free(reply);
```

Comment:
nitpick: if reply is NULL, there's no need to call free

Reply:
My understanding is that the free(reply) call should be removed because reply can be either NULL or a struct and if !reply means it can be only NULL.

Is there anything preventing this PR (beyond this free(reply) change ?)